RTCError extends DOMException

henbos opened the pull request:
Fixes #2082 

This is a reupload of https://github.com/w3c/webrtc-pc/pull/2084 with respec issues addressed.


<!--
    This comment and the below content is programatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/henbos/webrtc-pc/pull/2089.html" title="Last updated on Jan 31, 2019, 10:44 AM UTC (658a4f3)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webrtc-pc/2089/3455fd8...henbos:658a4f3.html" title="Last updated on Jan 31, 2019, 10:44 AM UTC (658a4f3)">Diff</a>

henbos commented:
Comment from @aboba from the old version of this PR:

> Section 11.1.2 says that sdpLineNumber is only set if the errorDetail value is "sdp-syntax-error". Would it make sense for sdpLineNumber to be set for other errorDetail values such as "hardware-encoder-not-available"/"hardware-encoder-error" (to indicate the specific encoder that is not available)?

henbos commented on webrtc.html:
```diff
@@ -11342,148 +11342,170 @@ <h3>Call Flow Browser to MCU</h3>
+          [
+              Exposed=Window,
+              Constructor(DOMString message, RTCErrorInit init)
+          ] interface RTCError /*: DOMException*/ {
```
I get a respec error if I do ": DOMException" and I don't know why. DOMException is an interface and you should be able to extend it. Commented it out to make respec happy.

aboba replied:
You get an error because DOMException is not defined in the document.

alvestrand replied:
You probably have to add "DOMException is defined in <ref>" to the terminology section up front.

henbos replied:
In other places in the spec, including in WebIDL (example: `callback RTCPeerConnectionErrorCallback = void (DOMException error);`), it doesn't complain about it and the link produce takes you to the interface definition of another spec.

Our list of terminology has 
"When referring to exceptions, the terms throw and create are defined in [WEBIDL-1]."
which is referring to a document whose DOMException is old: https://www.w3.org/TR/2016/REC-WebIDL-1-20161215/#dfn-DOMException.

This is the reference we want: https://heycam.github.io/webidl/#idl-DOMException

henbos replied:
Should we update the Terminology section in a separate PR or in this one? Our WebIDL reference might affect other references in the document, and I assume we don't want to have multiple references to different versions of WebIDL.

alvestrand replied:
Definitely need to consult @dontcallmedom, and we should do it in a separate PR.

henbos commented:
Reply to previously unaddressed comment:

> Section 11.1.2 says that sdpLineNumber is only set if the errorDetail value is "sdp-syntax-error". Would it make sense for sdpLineNumber to be set for other errorDetail values such as "hardware-encoder-not-available"/"hardware-encoder-error" (to indicate the specific encoder that is not available)?

Not sure, this is something that can happen at setParameters(), now when you're parsing SDP. I think having more info on that is a separate issue than this "refactoring" of RTCError.
